### PR TITLE
Adds Algolia docs search

### PR DESCRIPTION
**Pull request by ryanmoran**
## Summary

Implements #262.

⚠️ NOTE: the search box itself is not styled to match the mockup. This is just a basically functional implementation.

It is also worth noting that Algolia has setup some [configuration](https://github.com/algolia/docsearch-configs/blob/master/configs/paketo.json) on their end to scrape data from the site. This configuration would need to be updated if we made structural changes to the site that broke their scraping process.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by ForestEckhardt** [comment]
I am dubious to merge this without the styling. Did you want this merged as is or are you planning on adding styling before merge?

**Comment by fg-j** [comment]
@ForestEckhardt moving to draft -- let's not merge without styling

File: assets/js/shims.js
Comment (fg-j) [inline assets/js/shims.js:1]: Is this addition required for the Algolia search?

Comment (ryanmoran) [reply assets/js/shims.js:1]: Yeah. Its because they have some code that thinks its running on Node and not in a browser. This just mocks out the `process` global so that code can load and then ultimately do nothing.

**Comment by fg-j** [comment]
@ForestEckhardt I'll leave this to you to take a look at since @ryanmoran and I have both worked on this.

File: config.toml
Comment (ForestEckhardt) [inline config.toml:117]: Should this API key be exposed like this?

Comment (fg-j) [reply config.toml:117]: @ryanmoran, correct me if I'm wrong, but my understanding is that's the *public* search API key. Safe to be checked in. All it lets you do is query our search backend.

Comment (ryanmoran) [reply config.toml:117]: Yes. This is not a secret and so can be checked in.

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**
